### DISCUSSION TITLE
Converting an image file to an array

### DISCUSSION THREAD

**Discussion by Bodmer** [Bodmer/TFT_eSPI#1843] [General]
Image files can be converted to arrays using the tool here:
https://notisrac.github.io/FileToCArray/
To use this tool:
  1. Drag and drop file on "Choose file..." button
  2. Tick box "Treat as binary"
  3. Click "Convert"
  4. Click "Save as file" and move the header file to sketch folder
  5. Open the sketch in IDE
  6. Include the header file containing the array

This method can be used to convert jpeg and png image files to arrays. For an example see Flash_PNG example sketch. The smooth font vlw files can also be converted to arrays using the same method.

The instructions above create a complete byte by byte copy of the original file without any format conversion (i.e. a jpeg or png image file is NOT converted to a bitmap file).

**Comment by Prathamesh08glovatrix** [comment]
"lcd-image-converter "can we use this software? 
if yes please can you tell me the process?
thanks.

**Reply by Bodmer** [reply]
I have not used that software so cannot advise.

**Reply by eldar1973** [reply]
In lcd-image-converter set the parameters:
Settings - Conversion:
On the Image tab, check:
- Preset Color R8G8B8
- Data block 8 bits
- Byte order: Big-Endian
- Prefix: 0x
- Separator: ,
